### ISSUE TITLE
[LLMObs] Send llmobs events split by chunk if it's large enough

### ISSUE THREAD

**Issue by kimwonj77** [DataDog/dd-trace-py#10212]
### Summary of problem
As we're using batch LLM queue via celery. The LLM calls are huge and LLMObs endpoint refuses with 413 or Timed out.
We're using 100% sample rate but it became meaning less. (For price calcuation and etcs...)

Also, I saw d7203df0d6407316d46e8edefd3e8e6be30d72d0 and seems this is not the case (Because, each span is <1MB but Total Span is too large. because it sends at once)
I think spilt events are ok for library.

### Which version of dd-trace-py are you using?
`ddtrace==2.9.3`
### Which version of pip are you using?
`pip 24.1.2`

### Which libraries and their versions are you using?
```
celery==5.4.0
gevent==24.2.1
```
...

### How can we reproduce your problem?
Do a bunch of llm calls at once in one application. I'm using celery with gevent pool, 500 workers.

### What is the result that you get?
spans are sent to datadog

### What is the result that you expected?
Many spans are ignored with errors

#### Example Error:
<details>
<summary>Timed out</summary>
```
failed to send 810 LLMObs span events to llmobs-intake.datadoghq.com, 2 additional messages skipped<br/>
Traceback (most recent call last):<br/>
  File "/[redacted]/.venv/lib/python3.11/site-packages/ddtrace/llmobs/_writer.py", line 100, in periodic<br/>
    conn.request("POST", self._endpoint, enc_llm_events, self._headers)<br/>
  File "/[redacted]/lib/python3.11/http/client.py", line 1303, in request<br/>
    self._send_request(method, url, body, headers, encode_chunked)<br/>
  File "/[redacted]/lib/python3.11/http/client.py", line 1349, in _send_request<br/>
    self.endheaders(body, encode_chunked=encode_chunked)<br/>
  File "/[redacted]/lib/python3.11/http/client.py", line 1298, in endheaders<br/>
    self._send_output(message_body, encode_chunked=encode_chunked)<br/>
  File "/[redacted]/lib/python3.11/http/client.py", line 1097, in _send_output<br/>
    self.send(chunk)<br/>
  File "/[redacted]/lib/python3.11/http/client.py", line 1019, in send<br/>
    self.sock.sendall(data)<br/>
...
```
</details>

<details>
<summary>413 Entity Too Large</summary>
```
failed to send 768 LLMObs span events to https://llmobs-intake.datadoghq.com/api/v2/llmobs, got response code 413, status: b'{"errors":[{"status":"413","title":"Request Entity Too Large","detail":"Request too large"}]}'
```
</details>

**Comment by emmettbutler** [comment]
Thanks for the report, @kimwonj77. cc @Yun-Kim

**Comment by Yun-Kim** [comment]
Hi @kimwonj77, thanks for reaching out!

I believe #10093 actually addresses this issue and should be released in ddtrace 2.12 in the next couple weeks. I'll close this for now, but please re-open this issue if ddtrace 2.12 does not resolve this!

**Comment by kimwonj77** [comment]
Hi, I recently tested version 2.12 and noticed that the 'error log' is no more on logs. However, we are still only receiving the most recent LLM calls. We should be receiving at least 200 LLM calls span, but we are only getting around 30, and they all appear to be the latest ones. 

It’s possible that ddtrace may not be compatible with the gevent pool through Celery. (We are using a gevent pool with 250 workers for the LLM calls, as they are essentially just send-and-wait operations.)
